src/Compilers/Core/Portable/InternalUtilities/AdditionalTextComparer.cs

```diff
@@ -32,21 +32,42 @@ public bool Equals(AdditionalText? x, AdditionalText? y)
                 return false;
             }
 
-            var xText = x.GetText();
-            var yText = y.GetText();
+            try
+            {
+                var xText = GetTextOrNullIfBinary(x);
+                var yText = GetTextOrNullIfBinary(y);
+
+                if (xText is null || yText is null || xText.Length != yText.Length)
+                {
+                    return false;
```

Comment:
❗ If `xText` and `yText` are both null, then the additional text is _observably_ not changed and can be treated as equal. We should handle this case specifically by returning `true`.